### PR TITLE
77 anonymous posts and comments

### LINKED ISSUE #77: Anonymous Posts
Allow a user to create anonymous posts / comments since a user may not want to have their question associated back to them. and add in the ability to mark a post as read.

### PR DESCRIPTION
**Pull request by MarshallAsch**
closes #77. Added the ability to create anonymous posts and comments, It will not display the name of the author if it is flagged as anonymous. This will continue to work on a backwards compatible way. When an anonymous post / comment is on a device that does not support anonymous posts then the author will be visible. Old posts are forwards compatible.

If you are creating a new comment and you are the author of the original anonymous post then the comment will be default set to be anonymous.

### CODE REVIEW

File: app/src/main/java/ca/marshallasch/veil/utilities/Util.java [hunk 1]
```diff
@@ -168,17 +186,48 @@ public static String generateHash(byte[] data) {
...
+     * Create a new comment object, that is not assigned to a specific post.
+     * Note that this does not set the postID field, and the hash is made with the field unset.
+     *
+     * @param message the body of the comment
+     * @param author the author who wrote the comment
+     * @param postHash the postID that it belongs to
+     * @param anonymous if the post is anonymous or not
+     * @return a comment object with the postID field set
+     */
+    public static DhtProto.Comment createComment(@NonNull String message, @NonNull DhtProto.User author, @Nullable String postHash, boolean anonymous) {
```
Comment (weihanli101) [inline app/src/main/java/ca/marshallasch/veil/utilities/Util.java]: Maybe write some tests to see if the expected result will be set for more edge case scenarios especially involving the 'anonymous' param.
